### PR TITLE
Add file name info to SequentialFileReader.

### PR DESCRIPTION
**Pull request by riversand963**
We potentially need this information for tracing, profiling and diagnosis.

Test Plan:
```
$ make -j16 all check
```

### CODE REVIEW

File: db/db_impl.cc [hunk 1]
```diff
@@ -2368,7 +2368,7 @@ Status DBImpl::GetDbIdentity(std::string& identity) const {
     if (!s.ok()) {
       return s;
     }
-    id_file_reader.reset(new SequentialFileReader(std::move(idfile)));
+    id_file_reader.reset(new SequentialFileReader(std::move(idfile), idfilename));
```
Comment (siying) [inline db/db_impl.cc]: 80-char?

File: util/file_reader_writer.h [hunk 2]
```diff
@@ -26,11 +26,13 @@ std::unique_ptr<RandomAccessFile> NewReadaheadRandomAccessFile(
...
  private:
   std::unique_ptr<SequentialFile> file_;
+  std::string file_name_;
   std::atomic<size_t> offset_;  // read offset
 
  public:
-  explicit SequentialFileReader(std::unique_ptr<SequentialFile>&& _file)
-      : file_(std::move(_file)), offset_(0) {}
+  explicit SequentialFileReader(std::unique_ptr<SequentialFile>&& _file,
+                                const std::string file_name)
```
Comment (siying) [inline util/file_reader_writer.h]: I beblieve some compiler will complain it has the same name as file_name(). Rename it.

File: util/testutil.cc [hunk 3]
```diff
@@ -135,9 +135,9 @@ RandomAccessFileReader* GetRandomAccessFileReader(RandomAccessFile* raf) {
                                     "[test RandomAccessFileReader]");
 }
 
-SequentialFileReader* GetSequentialFileReader(SequentialFile* se) {
+SequentialFileReader* GetSequentialFileReader(SequentialFile* se, const std::string& fname) {
```
Comment (siying) [inline util/testutil.cc]: make format

File: db/db_impl.cc (hunk 1 above)
Comment (riversand963) [reply db/db_impl.cc]: Done.

File: util/file_reader_writer.h (hunk 2 above)
Comment (riversand963) [reply util/file_reader_writer.h]: Done.

File: util/testutil.cc (hunk 3 above)
Comment (riversand963) [reply util/testutil.cc]: Done.

**Pull request closed without merging**